PISTON-1085: Do not overwrite a system error message if its set when adding the error to a crossbar context

bradfordben opened the pull request:
Kazoo 4.3 PR https://github.com/2600hz/kazoo/pull/6700

bradfordben committed:
PISTON-1085: Do not overwrite a system error message if its set when
adding the error to a crossbar context

jamesaimonetti commented:
@bradfordben sorry for the delay, what's the API call that you've seen the overwrite happening on? What's the expected vs actual results of that API call? Would like to get a properly test case showing the issue.

jamesaimonetti committed:
Ignore deity module

jamesaimonetti commented:
@bradfordben would like to get a test for this; what API request triggers it?

kdamas93 committed:
KCRO-148: standardize handling of different types of UI apps (#374)

The purpose of this API is to replicate the behavior of the apps_store
API with the following differences:

- It will be possible to perform actions on the following resources
through the cb_application endpoint
- application document which includes screenshots, icons, entitlement,
and blocklist among others.

The creation and management of permissions for this will be in each
account, unlike the current scheme that was only allowed in the master
account.

This API will also allow handling different types of applications, for
example, commland_ui and monster_ui, a field has been added which
specifies the type of application that has been created. With this it
is hoped to be able to have controlled and grouped the different
applications that may arise within their category.

It has created its own module in core to execute actions from the
console, the module is called kazoo_ui_apps_maintenance, and the
related pr is 2600hz/kazoo-core#524, it has been developed based on
kazoo_documents_maintenance, keeping the functionality that exists for
monster-ui applications that follow the current architecture.

icehess committed:
KCRO-181: ignore empty mms body while there is a atleast a file (#463)

This will remove minimum from MMS schema body field and improves the
cb_mms to make sure we ignore empty body ONLY when there is at least a
file in the request.

If no file and empty body then the request fails as before with
validation error.

Commit:
KZOO-301: Add a period of user lockout for failed login attempts (#467)

Add simple account lock-out mechanism using Token Buckets if any of
the account's users fail to login with correct creds in a configurable
number of attempts.

jamesaimonetti committed:
tinkering with rebar3 as alternative experimental build system (#499)

This introduce rebar3 as an alternative build system, only for use
locally in your dev env.

Current Makefile system is robust and gets the job done. But it
increasingly getting difficult to run this in different env and OS,
and it is ridiculously SLOW. Average time is something around
7mins. If you run in VM it may be more close to 10min. If you add
macOS in equation, it is infinity (something around 25-3m min).

With Rebar3 the build time is something under or around 2min, for
everything, including deps.

Currently this is just a simple configuration to test out
rebar. `xref`, `dialyzer`, test and other targets are not tested, and
are not properly configured.

This tries to not depend on calling Makefile targets as much as
possible. This reduce the problems on different platforms especially
windows and macos. Some scripts was added to replace some tasks needed
to setup files for compilation.

jamesaimonetti committed:
KZOO-397: update schema for expected format of classifiers (#494)

Include required classifier properties

Commit:
PS-137: Doorman callflow schema updates (#496)

Commit:
KPUS-6: add push-type to AMQP payload (#495)

icehess committed:
bump orb to use custom couchdb image (#498)

kusalfernando committed:
KCRO-183: multiple functions to handle all possible scenarios for direction (#481)

* break direction/2 into multiple functions to handle all possible scenarios

* use proper function names and add comments to clarify

---------

Co-authored-by: Hesaam Farhang <hesaam@2600hz.com>

kdamas93 committed:
KCRO-188: Update contact lists (#500)

The purpose of this PR is to improve/adapt the current functionality
of cb_list to be able to save contact lists depending on the needs,
with these changes it will be possible to associate owner_id to the
lists so that they can later be filtered by user.

With this owner_id permission, validation logic will be added, about
who can access the lists.

Multiple entries can be associated to a list type document, each entry
will be a contact list, to achieve this it is necessary to update the
current schema (list_entry), the documentation and its respective
accessors.

It is also necessary to perform a refactor on the current code because
there are functions/calls that do not follow our current programming
patterns.

icehess committed:
update orb to consider os-release in cache keys (#502)

With upcoming changes, we need another variable to invalidate caches.

This adds /etc/os-release, uname -pmo and others together and uses them
as cache key in CI.

kdamas93 committed:
kcro-148:  allow admin to list available apps to their account (#506)

* KCRO-148: allow admin to list available apps to their account

This enables admin to see what is available so they can create
entitlement for it which enables the app in their account.

* update doc

---------

Co-authored-by: Hesaam Farhang <hesaam@2600hz.com>

Navoda626 committed:
KCRO-190: Additional APIs to consume pvt_chat.getstream in user doc (#505)

* Adding enable getstream feature and removing getstream feature APIs

* Adding APIs for geting user getstream enabled status and account getstream enabled users

* changes after ci checks

* Added documentation and removing the API for fetching getstream status for an user since its covered by user fetch by id API

* adding API to get the user getstream status

* Adding getstream accessor functions and changing get user getstream to pass the getstream obj

* enable adding json body for the getstream object for the PUT request

* Updating the DELETE method to remove getstream obj and pass the removed obj

* Adding patch method for getstream

* After circle-ci tests

* removing unwanted config file of teammate application

* after circle-ci checks

* refactor folds into manual recursion loops

* after code review

* adding meaningful error message

* resolving code-checks

* handling response codes from getstream API

* fetch getstream enabled users using crossbar_listing and erlang filtering

* removing unwanted log

* code cleanups

* code cleanups

---------

Co-authored-by: Navoda626 <navoda@aio.kazoo150.com>
Co-authored-by: James Aimonetti <james@2600hz.com>

jamesaimonetti committed:
AXS-101: schema updates (#514)

jamesaimonetti committed:
Schema updates from other public apps' cb modules

Navoda626 committed:
KCRO-189: schema updates for whitelabel (#508)

Adding new commland object and parameters in whitelabel schema

kdamas93 committed:
KCRO-194: Fix the double URL encoding (#518)

kdamas93 committed:
KZOO-410: UUID treatment for command selector (#516)

Co-authored-by: Hesaam Farhang <icehess@users.noreply.github.com>

kdamas93 committed:
KCRO-148: Updating docs with the correct path (#520)

Fixing patch of an app to update the document correctly

manushiperera committed:
KCRO-195 : As a user I want to update data from the Monster UI Marketplace connector (#522)

* Previously it was unable to set is_aio_cluster configuration through PATCH apps_store/marketplace API. Now this configuration can be set by PATCH apps_store/marketplace: {action:enable, settings:[is_aio_cluster,…]}. settings is an optional parameter.
A new market action is defined to handle the settings payload. Cluster name is added to response payload and cluster name can be updated via marketplace connector.

* explicitly handle updating local vs remote settings

* update docs, add more

* Fixed functionality issues, CI issues and typos
Remove api_url market action as it is depreciated

---------

Co-authored-by: Hesaam Farhang <hesaam@2600hz.com>
Co-authored-by: Hesaam Farhang <icehess@users.noreply.github.com>

kdamas93 committed:
KCRO-197: whitelabeled/blacklisted apps breaks the app selector list (#523)

Adjusting instructions for managing applications that come from the service plan
When a service plan has a whitelabeled/blacklisted application generates a break since the structure returned is not what was expected, this exception is adapted with the added changes
error reported :
https://2600hz-commercial.atlassian.net/browse/CAS-110

dhuynh-Kazoo committed:
KZOO-301: Set infinity for cb_conference_auth for account level (#527)

harenson committed:
DOC: Add guide to setup a local S3-compatible service MinIO (#530)

sanjeevie6353 committed:
Merge branch 'master' into dont-overwrite-system-error-message

manushiperera commented:
Hello @bradfordben,

Any update on this?

bradfordben commented:
Hi, sorry about the delay on getting back to you. This was so long ago i cant quite remember but i feel its something like if you try and update or create a object through crossbar and it has multiple validation errors it will now return all the errors in one json response rather than only the last error. I dont have a working setup at the moment to just try this quickly unfortunately.
